dataframe-jdbc/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/readJdbc.kt

```diff
@@ -0,0 +1,635 @@
+package org.jetbrains.kotlinx.dataframe.io
+
+import io.github.oshai.kotlinlogging.KotlinLogging
+import org.jetbrains.kotlinx.dataframe.AnyFrame
+import org.jetbrains.kotlinx.dataframe.DataFrame
+import org.jetbrains.kotlinx.dataframe.api.toDataFrame
+import org.jetbrains.kotlinx.dataframe.impl.schema.DataFrameSchemaImpl
+import org.jetbrains.kotlinx.dataframe.io.db.*
+import org.jetbrains.kotlinx.dataframe.schema.DataFrameSchema
+import java.sql.*
+import java.util.*
+
+private val logger = KotlinLogging.logger {}
+
+/**
+ * The default limit value.
+ *
+ * This constant represents the default limit value to be used in cases where no specific limit
+ * is provided.
+ *
+ * @see Int.MIN_VALUE
+ */
+private const val DEFAULT_LIMIT = Int.MIN_VALUE
+
+/**
+ * Represents a column in a database table to keep all required meta-information.
+ *
+ * @property [name] the name of the column.
+ * @property [sqlType] the SQL data type of the column.
+ * @property [jdbcType] the JDBC data type of the column.
+ * @property [size] the size of the column.
+ */
+public data class TableColumnMetadata(val name: String, val sqlType: String, val jdbcType: Int, val size: Int)
+
+/**
+ * Represents a table metadata to store information about a database table,
+ * including its name, schema name, and catalogue name.
+ *
+ * NOTE: we need to extract both, [schemaName] and [catalogue]
+ * because the different databases have different implementations of metadata.
+ *
+ * @property [name] the name of the table.
+ * @property [schemaName] the name of the schema the table belongs to (optional).
+ * @property [catalogue] the name of the catalogue the table belongs to (optional).
+ */
+public data class TableMetadata(val name: String, val schemaName: String?, val catalogue: String?)
+
+/**
+ * Represents the configuration for a database connection.
+ *
+ * @property [url] the URL of the database.
+ * @property [user] the username used for authentication (optional, default is empty string).
+ * @property [password] the password used for authentication (optional, default is empty string).
+ */
+public data class DatabaseConfiguration(val url:String, val user: String = "", val password: String = "")
+
+/**
+ * Reads data from an SQL table and converts it into a DataFrame.
+ *
+ * @param [dbConfig] the configuration for the database, including URL, user, and password.
+ * @param [tableName] the name of the table to read data from.
+ * @return the DataFrame containing the data from the SQL table.
+ */
+public fun DataFrame.Companion.readSqlTable(dbConfig: DatabaseConfiguration, tableName: String): AnyFrame {
```

Comment:
Data class is easier to reuse imo. Like, i tried to experiment with SQLite db: read multiple tables, made multiple `readSqlTable` calls.  With data class i created a variable. With DSL it would be more tiresome, especially in notebook where there're no refactorings (?) like extract

Reply:
Easier to reuse yes, but not easier to type when calling the function. Maybe we could both support a DatabaseConfiguration and just arguments in the function call?

Reply:
Again, for now, a lot of functions, we tracked all the ideas here or I created a separate issues for that.

I want to give users ability to taste it with the status experimental to handle more cases